### PR TITLE
#58 taking issues 1-5

### PR DESCRIPTION
**Pull request by ja6a**

### CODE REVIEW

File: skytap/data_source_skytap_template.go [hunk 1]
```diff
@@ -67,8 +67,10 @@ func dataSourceSkytapTemplateRead(d *schema.ResourceData, meta interface{}) erro
 		template = templates[0]
 	}
 
-	d.SetId(*template.ID)
-	d.Set("name", *template.Name)
+	templateID := *template.ID
```
Comment (pegerto) [inline skytap/data_source_skytap_template.go]: what if template.Id or templatNAme is null

File: skytap/resource_skytap_environment.go [hunk 2]
```diff
@@ -145,7 +135,8 @@ func resourceSkytapEnvironmentCreate(d *schema.ResourceData, meta interface{}) e
 		return errors.Errorf("error creating environment: %v", err)
 	}
 
-	d.SetId(*environment.ID)
+	environmentID := *environment.ID
```
Comment (pegerto) [inline skytap/resource_skytap_environment.go]: can enviroment been null, can we validate this.

File: skytap/resource_skytap_network.go [hunk 3]
```diff
@@ -92,7 +92,8 @@ func resourceSkytapNetworkCreate(d *schema.ResourceData, meta interface{}) error
 		return errors.Errorf("error creating network: %v", err)
 	}
 
-	d.SetId(*network.ID)
+	networkID := *network.ID
```
Comment (pegerto) [inline skytap/resource_skytap_network.go]: same if we check for dereferencing, then we should check if network is null.

File: skytap/resource_skytap_vm.go [hunk 4]
```diff
@@ -136,8 +129,8 @@ func resourceSkytapVMCreate(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return errors.Errorf("error creating vm: %v", err)
 	}
-
-	d.SetId(*vm.ID)
+	vmID := *vm.ID
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go]: safe check

File: skytap/data_source_skytap_template.go (hunk 1 above)
Comment (ja6a) [reply skytap/data_source_skytap_template.go]: changed to nil check

File: skytap/resource_skytap_network.go (hunk 3 above)
Comment (ja6a) [reply skytap/resource_skytap_network.go]: changed to nil check

File: skytap/resource_skytap_vm.go (hunk 4 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go]: changed to nil check

**Comment by ja6a** [comment]
Also:
1) added nil checks to other pointer dereferences. 
2) added validation method for project roles.

**Comment by pegerto** [comment]
no functional change, only refactoring.

**Pull request merged**